### PR TITLE
Provide Solace Queue Scaler

### PR DESCRIPTION
**Pull request by dennis-brinley**
Signed-off-by: Dennis Brinley <dennis.brinley@solace.com>
Includes solace_scaler Unit tests, end-to-end tests, and mod to handler
** I'm waiting for sign-off for the documentation. Will commit early next week.

By making a contribution to this project, I certify that:

    (a) The contribution was created in whole or in part by me and I have the right to submit it under the open source license indicated in the file; or

    (b) The contribution is based upon previous work that, to the best of my knowledge, is covered under an appropriate open source license and I have the right under that license to submit that work with modifications, whether created in whole or in part by me, under the same open source license (unless I am permitted to submit under a different license), as indicated in the file; or

    (c) The contribution was provided directly to me by some other person who certified (a), (b) or (c) and I have not modified it.

    (d) I understand and agree that this project and the contribution are public and that a record of the contribution (including all personal information I submit with it, including my sign-off) is maintained indefinitely and may be redistributed consistent with this project or the open source license(s) involved.

### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))
- [x] Tests have been added
- [x] A PR is opened to update our Helm chart ([repo](https://github.com/kedacore/charts)) *(if applicable, ie. when deployment manifests are modified)*
- [x] A PR is opened to update the documentation on (https://github.com/kedacore/keda-docs/pull/486) *(if applicable)*
- [x] Changelog has been updated

Fixes #

### CODE REVIEW

**Comment by zroubalik** [comment]
Thanks, there are a couple of linting problems and some failing test, could you please fix that?

**Comment by dennis-brinley** [comment]
Corrected

File: pkg/scalers/solace_scaler.go [hunk 1]
```diff
@@ -0,0 +1,367 @@
...
+			Target: v2beta2.MetricTarget{
+				Type:         v2beta2.AverageValueMetricType,
+				AverageValue: targetMetricValue,
+			},
+		}
+		metricSpec := v2beta2.MetricSpec{External: externalMetric, Type: solaceExtMetricType}
+		metricSpecList = append(metricSpecList, metricSpec)
+	}
+	return metricSpecList
+}
```
Comment (zroubalik) [inline pkg/scalers/solace_scaler.go:270]: @coderanger do you think that the similar approcah would be beneficial for Rabbit MQ scaler?

**Comment by tomkerkhove** [comment]
Thank you for your contribution @dennis-brinley! I'll take a look at the docs asap!

File: pkg/scalers/solace_scaler.go [hunk 2]
```diff
@@ -0,0 +1,367 @@
...
+	}
+	return &meta, nil
+}
+
+func getSolaceSempCredentials(config *ScalerConfig) (u string, p string, err error) {
+	//	GET CREDENTIALS
+	//	The username must be a valid broker ADMIN user identifier with read access to SEMP for the broker, VPN, and relevant objects
+	//	The scaler will attempt to acquire username and then password independently. For each:
+	//	- Search K8S Secret (Encoded)
+	//	- Search environment variable specified by config at 'usernameEnv' / 'passwordEnv'
```
Comment (zroubalik) [inline pkg/scalers/solace_scaler.go:196]:
```suggestion
	//	- Search environment variable specified by config at 'usernameFromEnv' / 'passwordFromEnv'
```
Minor nit, we are using `fromEvn` suffix for this kind of variables

**Review by zroubalik — CHANGES_REQUESTED** [review]
Looking good, +100 for thorough test coverage.

Minor nit in metadata from env variables naming.

File: pkg/scalers/solace_scaler.go (hunk 2 above)
Comment (tomkerkhove) [reply pkg/scalers/solace_scaler.go:196]: 🤖 _Did you mean `fromEnv`?_

Comment (zroubalik) [reply pkg/scalers/solace_scaler.go:196]: 🤦 oh boy...

Comment (dennis-brinley) [reply pkg/scalers/solace_scaler.go:196]: I'll put in a change - also need to update the doc.

File: pkg/scalers/solace_scaler.go (hunk 1 above)
Comment (dennis-brinley) [reply pkg/scalers/solace_scaler.go:270]: Do you need me to do anything with this or can I resolve?

Comment (zroubalik) [reply pkg/scalers/solace_scaler.go:270]: @dennis-brinley nope, this was just FYI for @coderanger :)

**Review by ahmelsayed — APPROVED** [review]

**Comment by ahmelsayed** [comment]
A small merge conflict in `CHANGELOG.md`

**Review by zroubalik — APPROVED** [review]

**Pull request merged**